tests/unit/ai/test_cursor_provider.py

```
@@ -0,0 +1,309 @@
+        input_text = call_kwargs.kwargs.get("input", "")
+        assert_that(input_text).contains("sys prompt")
+        assert_that(input_text).contains("user msg")
+
+
+def test_complete_raises_on_subprocess_timeout(provider):
+    """Raise AIProviderError when CLI times out."""
+    with patch(
+        "subprocess.run",
+        side_effect=subprocess.TimeoutExpired(cmd="agent", timeout=60),
+    ):
+        with pytest.raises(AIProviderError, match="timed out"):
```

Ruff (0.15.20)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)